File: parser/src/main/java/com/github/nsc/de/shake/parser/node/factor/StringNode.java
package com.github.nsc.de.shake.parser.node.factor;

import com.github.nsc.de.shake.parser.node.ValuedNode;
import com.github.nsc.de.shake.util.characterinput.position.PositionMap;

public class StringNode extends ValuedNode {

    private final String value;

    public StringNode(PositionMap map, String value) {
        super(map);
        this.value = value;
    }

    public String getValue() {
        return value;
    }

}

File: parser/src/main/kotlin/com/github/nsc/de/shake/parser/node/factor/StringNode.kt
package com.github.nsc.de.shake.parser.node.factor

import com.github.nsc.de.shake.parser.node.ValuedNode
import com.github.nsc.de.shake.util.characterinput.position.PositionMap

class StringNode(map: PositionMap, val value: String) : ValuedNode(map) {
    override fun toJson(): Map<String, *> = mapOf("name" to "StringNode", "number" to value)
}